### ISSUE TITLE
unhandled promise rejection if no wallet balance

### ISSUE THREAD

**Issue by RnbWd** [Giveth/giveth-dapp#389]
if the wallet is locked, and you try to create a new campaign or community, when the the swal popup is confirmed, the Dapp throws an unhandled promise rejection error

**Comment by RnbWd** [comment]
I'm attempting to add a similar a function in middleware.jsx for handling when someone isn't authenticated or logged in, and I ran into this error while testing (quite fitting), so I'll try and tackle both simultaneously.

**Comment by vojtechsimetka** [comment]
Thank you @RnbWd I was eyeing that one. Perfect if you can tackle it!

**Comment by RnbWd** [comment]
basically we need to add catch statements for every promise that returns a rejection so that the app doesn't crash during development, even if we aren't doing anything with the error at the moment.

**Comment by vojtechsimetka** [comment]
I mean there is a React solution for that. It would be great if you used it and created at least the top level boundary https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html#introducing-error-boundaries

**Comment by RnbWd** [comment]
React's error component doesn't catch errors for promises that we generate ourselves... I just tested it.  That method only works for "catching errors during rending, lifecycle methods, and in the constructors of react components". 

I added catch statements to checkWalletBalance throughout the codebase to prevent the app from crashing during testing. It makes sense to handle the errors consistently throughout the app, but its important to have the catch statements for using the app during development.
